include/odp/api/spec/proto_stats.h

```diff
@@ -25,14 +25,67 @@ extern "C" {
  * Invalid proto stats handle
  */
 
+/** ODP proto statistics opt for Tx
+ *
+ * Statistics that can be enabled in proto stats objects for Tx of a packet via
+ * Packet IO. Pktout config `odp_pktout_config_opt_t::bit::proto_stats_ena`
+ * needs to be enabled for the offload to work.
+ *
+ * Tx packet and octet sent/drop statistics might include packets sent/dropped via
+ * Traffic Manager or Tx packet Aging or due to any other Tx errors. It is
+ * implementation specific as to what all Tx sent/drop events are accounted for.
+ */
+typedef union odp_proto_stats_tx_opt_t {
+	/** Option flags */
+	struct {
+		/** Packet sent count */
+		int64_t pkts : 1;
+
+		/** Packet drop count */
+		uint64_t pkt_drops : 1;
+
+		/** Packet sent Octet counter 0 */
+		uint64_t oct_count0 : 1;
+
+		/** Packet drop Octet counter 0 */
+		uint64_t oct_count0_drop : 1;
+
+		/** Packet sent octet counter 1 */
+		uint64_t oct_count1 : 1;
+
+		/** Packet drop octet counter 1 */
+		uint64_t oct_count1_drop : 1;
+	} bit;
+
+	/** All bits of the bit field structure
+	 *
+	 * This field can be used to set/clear all flags, or bitwise
+	 * operations over the entire structure.
+	 */
+	uint64_t all_bits;
+} odp_proto_stats_tx_opt_t;
```

Comment:
I'd prefer that the options/capa would have same naming convention (/hierarchy) that the counter data struct. Since odp_proto_stats_data_t is flat, this could be simply e.g. odp_proto_stats_counters_t (or odp_proto_stats_counter_flags_t ,etc) and contain all counter flags flat (tx_pkts, tx_pkt_drops, ...).

If this struct will just list all counters (and no other options), then "opt" may not be optimal name for it.